### ISSUE TITLE
Unable to check_cmd file: manage_file() takes at most 18 arguments (22 given)

### ISSUE THREAD

**Issue by anlutro** [saltstack/salt#38111]
file.managed with check_cmd seems broken in develop. I'm using salt-ssh but can't imagine it matters.

```
          ID: /etc/sudoers
    Function: file.managed
      Result: False
     Comment: Unable to check_cmd file: manage_file() takes at most 18 arguments (22 given)
     Started: 21:50:03.485034
    Duration: 12.335 ms
     Changes:  
```

```
/etc/sudoers:
  file.managed:
    - source: salt://sudo/sudoers.jinja
    - template: jinja
    - context:
        rules: {{ pillar.get('sudoers', {}) | json }}
        users: {{ sudo_users | json }}
    - user: root
    - group: root
    - mode: 440
    - check_cmd: visudo -c -s -f
```

```
Salt Version:
           Salt: 2016.11.0-469-g2a1066e
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.12 (default, Jun 28 2016, 06:57:42) [GCC]
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 16.0.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.4.2
            ZMQ: 4.1.5
 
System Versions:
           dist:   
        machine: x86_64
        release: 4.8.10-1-default
         system: Linux
        version: Not Installed
```

**Comment by Ch3LL** [comment]
@anlutro looks like i'm able to replicate this with a similar state:

```
/tmp/sudoers:
  file.managed:
    - user: root
    - group: root
    - mode: 440
    - check_cmd: visudo -c -s -f
```

Here is a docker container for whoever wants to quickly replicate it:

1. `docker run -it -v /home/ch3ll/git/salt:/testing/ ch3ll/issues:38111 salt-call --local state.sls test -ldebug` (where /home/ch3ll/git/salt is a local cloned git repo)

thanks this will need to get fixed up

**Comment by rallytime** [comment]
Thanks for reporting this issue @anlutro and for the excellent reproducible test case, @Ch3LL. I have fixed the issue in PR #39002. Using the same state that @Ch3LL provided, here is the output with the fix:
```
# salt-call --local state.sls test
local:
----------
          ID: /tmp/sudoers
    Function: file.managed
      Result: True
     Comment: Empty file
     Started: 17:54:11.345151
    Duration: 301.116 ms
     Changes:
              ----------
              mode:
                  0440
              new:
                  file /tmp/sudoers created

Summary for local
------------
Succeeded: 1 (changed=1)
Failed:    0
------------
Total states run:     1
Total run time: 301.116 ms
```
